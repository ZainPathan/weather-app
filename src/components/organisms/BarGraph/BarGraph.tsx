import React from 'react';
import {
    BarChart,
    Bar,
    XAxis,
    YAxis,
    CartesianGrid,
    Tooltip,
    Legend,
    ResponsiveContainer
} from 'recharts';

interface BarGraphProps {
    weatherData: any,
    currentCardDate: string,
    temperatureUnit: string
}

const BarGraph = ({weatherData, currentCardDate, temperatureUnit}: BarGraphProps) => {
    const graphData = weatherData && weatherData[currentCardDate].segments;

    return (weatherData &&
        <ResponsiveContainer width="100%" height="100%">
            <BarChart
                width={500}
                height={300}
                data={graphData}
                margin={{
                    top: 5,
                    right: 30,
                    left: 20,
                    bottom: 5,
                }}
            >
                <CartesianGrid strokeDasharray="3 3" />
                <XAxis
                    dataKey="time"
                />
                <YAxis />
                <Tooltip />
                <Legend />
                {temperatureUnit === 'celsius' && <Bar dataKey="temperatureInCelsius" fill="#3f50b5" />}
                {temperatureUnit === 'fahrenheit' && <Bar dataKey="temperatureInFahrenheit" fill="#f50057" />}
            </BarChart>
        </ResponsiveContainer>
    );
};

export default BarGraph;
