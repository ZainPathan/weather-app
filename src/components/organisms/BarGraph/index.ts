import BarGraph from './BarGraph';

export {
    BarGraph
};

export default BarGraph;