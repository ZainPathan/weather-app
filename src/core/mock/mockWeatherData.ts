export const mockWeatherData = {
    '19-07-2021': {
        list: {
            '00:00': {
                temp: 288.46,
                feels_like: 288.34,
                temp_min: 287.46,
                temp_max: 288.46,
                pressure: 1020,
                sea_level: 1020,
                grnd_level: 958,
                humidity: 88,
                temp_kf: 1,
                time: '00:00'
            },
            '03:00': {
                temp: 287.77,
                feels_like: 287.64,
                temp_min: 286.38,
                temp_max: 287.77,
                pressure: 1020,
                sea_level: 1020,
                grnd_level: 958,
                humidity: 90,
                temp_kf: 1.39,
                time: '03:00'
            },
            '06:00': {
                temp: 289.11,
                feels_like: 289.09,
                temp_min: 289.11,
                temp_max: 289.44,
                pressure: 1019,
                sea_level: 1019,
                grnd_level: 958,
                humidity: 89,
                temp_kf: -0.33,
                time: '06:00'
            },
            '09:00': {
                temp: 294.27,
                feels_like: 294.24,
                temp_min: 294.27,
                temp_max: 294.27,
                pressure: 1018,
                sea_level: 1018,
                grnd_level: 958,
                humidity: 69,
                temp_kf: 0,
                time: '09:00'
            },
            '12:00': {
                temp: 296.91,
                feels_like: 296.83,
                temp_min: 296.91,
                temp_max: 296.91,
                pressure: 1017,
                sea_level: 1017,
                grnd_level: 957,
                humidity: 57,
                temp_kf: 0,
                time: '12:00'
            },
            '15:00': {
                temp: 296.79,
                feels_like: 296.7,
                temp_min: 296.79,
                temp_max: 296.79,
                pressure: 1016,
                sea_level: 1016,
                grnd_level: 957,
                humidity: 57,
                temp_kf: 0,
                time: '15:00'
            },
            '18:00': {
                temp: 292.07,
                feels_like: 292.29,
                temp_min: 292.07,
                temp_max: 292.07,
                pressure: 1016,
                sea_level: 1016,
                grnd_level: 956,
                humidity: 87,
                temp_kf: 0,
                time: '18:00'
            },
            '21:00': {
                temp: 288.07,
                feels_like: 288.05,
                temp_min: 288.07,
                temp_max: 288.07,
                pressure: 1018,
                sea_level: 1018,
                grnd_level: 957,
                humidity: 93,
                temp_kf: 0,
                time: '21:00'
            }
        },
        segments: [
            {
                temperatureInCelsius: 15,
                temperatureInFahrenheit: 59,
                time: '00:00'
            },
            {
                temperatureInCelsius: 14,
                temperatureInFahrenheit: 57.2,
                time: '03:00'
            },
            {
                temperatureInCelsius: 16,
                temperatureInFahrenheit: 60.8,
                time: '06:00'
            },
            {
                temperatureInCelsius: 21,
                temperatureInFahrenheit: 69.8,
                time: '09:00'
            },
            {
                temperatureInCelsius: 23,
                temperatureInFahrenheit: 73.4,
                time: '12:00'
            },
            {
                temperatureInCelsius: 23,
                temperatureInFahrenheit: 73.4,
                time: '15:00'
            },
            {
                temperatureInCelsius: 19,
                temperatureInFahrenheit: 66.2,
                time: '18:00'
            },
            {
                temperatureInCelsius: 15,
                temperatureInFahrenheit: 59,
                time: '21:00'
            }
        ],
        temperature: 2333.4500000000003,
        minTemperature: 2331.06,
        maxTemperature: 2333.78,
        count: 8,
        date: '19-07-2021',
        dateString: '19 Jul 21'
    },
    '20-07-2021': {
        list: {
            '00:00': {
                temp: 286.8,
                feels_like: 286.7,
                temp_min: 286.8,
                temp_max: 286.8,
                pressure: 1018,
                sea_level: 1018,
                grnd_level: 957,
                humidity: 95,
                temp_kf: 0,
                time: '00:00'
            },
            '03:00': {
                temp: 285.79,
                feels_like: 285.62,
                temp_min: 285.79,
                temp_max: 285.79,
                pressure: 1018,
                sea_level: 1018,
                grnd_level: 957,
                humidity: 96,
                temp_kf: 0,
                time: '03:00'
            },
            '06:00': {
                temp: 287.78,
                feels_like: 287.62,
                temp_min: 287.78,
                temp_max: 287.78,
                pressure: 1019,
                sea_level: 1019,
                grnd_level: 958,
                humidity: 89,
                temp_kf: 0,
                time: '06:00'
            },
            '09:00': {
                temp: 292.71,
                feels_like: 292.42,
                temp_min: 292.71,
                temp_max: 292.71,
                pressure: 1019,
                sea_level: 1019,
                grnd_level: 958,
                humidity: 65,
                temp_kf: 0,
                time: '09:00'
            },
            '12:00': {
                temp: 295.53,
                feels_like: 295.21,
                temp_min: 295.53,
                temp_max: 295.53,
                pressure: 1017,
                sea_level: 1017,
                grnd_level: 958,
                humidity: 53,
                temp_kf: 0,
                time: '12:00'
            },
            '15:00': {
                temp: 295.89,
                feels_like: 295.58,
                temp_min: 295.89,
                temp_max: 295.89,
                pressure: 1017,
                sea_level: 1017,
                grnd_level: 958,
                humidity: 52,
                temp_kf: 0,
                time: '15:00'
            },
            '18:00': {
                temp: 292.11,
                feels_like: 292.07,
                temp_min: 292.11,
                temp_max: 292.11,
                pressure: 1018,
                sea_level: 1018,
                grnd_level: 958,
                humidity: 77,
                temp_kf: 0,
                time: '18:00'
            },
            '21:00': {
                temp: 287.6,
                feels_like: 287.4,
                temp_min: 287.6,
                temp_max: 287.6,
                pressure: 1021,
                sea_level: 1021,
                grnd_level: 959,
                humidity: 88,
                temp_kf: 0,
                time: '21:00'
            }
        },
        segments: [
            {
                temperatureInCelsius: 13,
                temperatureInFahrenheit: 55.4,
                time: '00:00'
            },
            {
                temperatureInCelsius: 12,
                temperatureInFahrenheit: 53.6,
                time: '03:00'
            },
            {
                temperatureInCelsius: 14,
                temperatureInFahrenheit: 57.2,
                time: '06:00'
            },
            {
                temperatureInCelsius: 19,
                temperatureInFahrenheit: 66.2,
                time: '09:00'
            },
            {
                temperatureInCelsius: 22,
                temperatureInFahrenheit: 71.6,
                time: '12:00'
            },
            {
                temperatureInCelsius: 22,
                temperatureInFahrenheit: 71.6,
                time: '15:00'
            },
            {
                temperatureInCelsius: 19,
                temperatureInFahrenheit: 66.2,
                time: '18:00'
            },
            {
                temperatureInCelsius: 14,
                temperatureInFahrenheit: 57.2,
                time: '21:00'
            }
        ],
        temperature: 2324.21,
        minTemperature: 2324.21,
        maxTemperature: 2324.21,
        count: 8,
        date: '20-07-2021',
        dateString: '20 Jul 21'
    },
    '21-07-2021': {
        list: {
            '00:00': {
                temp: 286.18,
                feels_like: 285.84,
                temp_min: 286.18,
                temp_max: 286.18,
                pressure: 1021,
                sea_level: 1021,
                grnd_level: 959,
                humidity: 88,
                temp_kf: 0,
                time: '00:00'
            },
            '03:00': {
                temp: 285.13,
                feels_like: 284.73,
                temp_min: 285.13,
                temp_max: 285.13,
                pressure: 1021,
                sea_level: 1021,
                grnd_level: 959,
                humidity: 90,
                temp_kf: 0,
                time: '03:00'
            },
            '06:00': {
                temp: 288.69,
                feels_like: 288.39,
                temp_min: 288.69,
                temp_max: 288.69,
                pressure: 1021,
                sea_level: 1021,
                grnd_level: 960,
                humidity: 80,
                temp_kf: 0,
                time: '06:00'
            },
            '09:00': {
                temp: 293.4,
                feels_like: 293.07,
                temp_min: 293.4,
                temp_max: 293.4,
                pressure: 1021,
                sea_level: 1021,
                grnd_level: 961,
                humidity: 61,
                temp_kf: 0,
                time: '09:00'
            },
            '12:00': {
                temp: 295.94,
                feels_like: 295.61,
                temp_min: 295.94,
                temp_max: 295.94,
                pressure: 1020,
                sea_level: 1020,
                grnd_level: 961,
                humidity: 51,
                temp_kf: 0,
                time: '12:00'
            },
            '15:00': {
                temp: 296.01,
                feels_like: 295.71,
                temp_min: 296.01,
                temp_max: 296.01,
                pressure: 1020,
                sea_level: 1020,
                grnd_level: 961,
                humidity: 52,
                temp_kf: 0,
                time: '15:00'
            },
            '18:00': {
                temp: 292.5,
                feels_like: 292.4,
                temp_min: 292.5,
                temp_max: 292.5,
                pressure: 1020,
                sea_level: 1020,
                grnd_level: 960,
                humidity: 73,
                temp_kf: 0,
                time: '18:00'
            },
            '21:00': {
                temp: 288.13,
                feels_like: 287.82,
                temp_min: 288.13,
                temp_max: 288.13,
                pressure: 1022,
                sea_level: 1022,
                grnd_level: 961,
                humidity: 82,
                temp_kf: 0,
                time: '21:00'
            }
        },
        segments: [
            {
                temperatureInCelsius: 13,
                temperatureInFahrenheit: 55.4,
                time: '00:00'
            },
            {
                temperatureInCelsius: 12,
                temperatureInFahrenheit: 53.6,
                time: '03:00'
            },
            {
                temperatureInCelsius: 15,
                temperatureInFahrenheit: 59,
                time: '06:00'
            },
            {
                temperatureInCelsius: 20,
                temperatureInFahrenheit: 68,
                time: '09:00'
            },
            {
                temperatureInCelsius: 22,
                temperatureInFahrenheit: 71.6,
                time: '12:00'
            },
            {
                temperatureInCelsius: 23,
                temperatureInFahrenheit: 73.4,
                time: '15:00'
            },
            {
                temperatureInCelsius: 19,
                temperatureInFahrenheit: 66.2,
                time: '18:00'
            },
            {
                temperatureInCelsius: 15,
                temperatureInFahrenheit: 59,
                time: '21:00'
            }
        ],
        temperature: 2325.98,
        minTemperature: 2325.98,
        maxTemperature: 2325.98,
        count: 8,
        date: '21-07-2021',
        dateString: '21 Jul 21'
    },
    '22-07-2021': {
        list: {
            '00:00': {
                temp: 286.42,
                feels_like: 286.1,
                temp_min: 286.42,
                temp_max: 286.42,
                pressure: 1022,
                sea_level: 1022,
                grnd_level: 960,
                humidity: 88,
                temp_kf: 0,
                time: '00:00'
            },
            '03:00': {
                temp: 285.32,
                feels_like: 284.94,
                temp_min: 285.32,
                temp_max: 285.32,
                pressure: 1021,
                sea_level: 1021,
                grnd_level: 960,
                humidity: 90,
                temp_kf: 0,
                time: '03:00'
            },
            '06:00': {
                temp: 288.84,
                feels_like: 288.5,
                temp_min: 288.84,
                temp_max: 288.84,
                pressure: 1021,
                sea_level: 1021,
                grnd_level: 960,
                humidity: 78,
                temp_kf: 0,
                time: '06:00'
            },
            '09:00': {
                temp: 293.93,
                feels_like: 293.63,
                temp_min: 293.93,
                temp_max: 293.93,
                pressure: 1020,
                sea_level: 1020,
                grnd_level: 960,
                humidity: 60,
                temp_kf: 0,
                time: '09:00'
            },
            '12:00': {
                temp: 296.52,
                feels_like: 296.27,
                temp_min: 296.52,
                temp_max: 296.52,
                pressure: 1019,
                sea_level: 1019,
                grnd_level: 960,
                humidity: 52,
                temp_kf: 0,
                time: '12:00'
            },
            '15:00': {
                temp: 296.76,
                feels_like: 296.53,
                temp_min: 296.76,
                temp_max: 296.76,
                pressure: 1018,
                sea_level: 1018,
                grnd_level: 959,
                humidity: 52,
                temp_kf: 0,
                time: '15:00'
            },
            '18:00': {
                temp: 293.16,
                feels_like: 293.15,
                temp_min: 293.16,
                temp_max: 293.16,
                pressure: 1018,
                sea_level: 1018,
                grnd_level: 958,
                humidity: 74,
                temp_kf: 0,
                time: '18:00'
            },
            '21:00': {
                temp: 288.99,
                feels_like: 288.82,
                temp_min: 288.99,
                temp_max: 288.99,
                pressure: 1018,
                sea_level: 1018,
                grnd_level: 957,
                humidity: 84,
                temp_kf: 0,
                time: '21:00'
            }
        },
        segments: [
            {
                temperatureInCelsius: 13,
                temperatureInFahrenheit: 55.4,
                time: '00:00'
            },
            {
                temperatureInCelsius: 12,
                temperatureInFahrenheit: 53.6,
                time: '03:00'
            },
            {
                temperatureInCelsius: 15,
                temperatureInFahrenheit: 59,
                time: '06:00'
            },
            {
                temperatureInCelsius: 20,
                temperatureInFahrenheit: 68,
                time: '09:00'
            },
            {
                temperatureInCelsius: 23,
                temperatureInFahrenheit: 73.4,
                time: '12:00'
            },
            {
                temperatureInCelsius: 23,
                temperatureInFahrenheit: 73.4,
                time: '15:00'
            },
            {
                temperatureInCelsius: 20,
                temperatureInFahrenheit: 68,
                time: '18:00'
            },
            {
                temperatureInCelsius: 15,
                temperatureInFahrenheit: 59,
                time: '21:00'
            }
        ],
        temperature: 2329.94,
        minTemperature: 2329.94,
        maxTemperature: 2329.94,
        count: 8,
        date: '22-07-2021',
        dateString: '22 Jul 21'
    },
    '23-07-2021': {
        list: {
            '00:00': {
                temp: 287.63,
                feels_like: 287.46,
                temp_min: 287.63,
                temp_max: 287.63,
                pressure: 1018,
                sea_level: 1018,
                grnd_level: 957,
                humidity: 89,
                temp_kf: 0,
                time: '00:00'
            },
            '03:00': {
                temp: 286.7,
                feels_like: 286.51,
                temp_min: 286.7,
                temp_max: 286.7,
                pressure: 1018,
                sea_level: 1018,
                grnd_level: 956,
                humidity: 92,
                temp_kf: 0,
                time: '03:00'
            },
            '06:00': {
                temp: 290.14,
                feels_like: 290.01,
                temp_min: 290.14,
                temp_max: 290.14,
                pressure: 1017,
                sea_level: 1017,
                grnd_level: 956,
                humidity: 81,
                temp_kf: 0,
                time: '06:00'
            },
            '09:00': {
                temp: 295.51,
                feels_like: 295.39,
                temp_min: 295.51,
                temp_max: 295.51,
                pressure: 1015,
                sea_level: 1015,
                grnd_level: 956,
                humidity: 61,
                temp_kf: 0,
                time: '09:00'
            },
            '12:00': {
                temp: 298.71,
                feels_like: 298.68,
                temp_min: 298.71,
                temp_max: 298.71,
                pressure: 1014,
                sea_level: 1014,
                grnd_level: 955,
                humidity: 52,
                temp_kf: 0,
                time: '12:00'
            },
            '15:00': {
                temp: 298.71,
                feels_like: 298.76,
                temp_min: 298.71,
                temp_max: 298.71,
                pressure: 1013,
                sea_level: 1013,
                grnd_level: 954,
                humidity: 55,
                temp_kf: 0,
                time: '15:00'
            },
            '18:00': {
                temp: 294.69,
                feels_like: 294.91,
                temp_min: 294.69,
                temp_max: 294.69,
                pressure: 1012,
                sea_level: 1012,
                grnd_level: 953,
                humidity: 77,
                temp_kf: 0,
                time: '18:00'
            },
            '21:00': {
                temp: 290.75,
                feels_like: 290.81,
                temp_min: 290.75,
                temp_max: 290.75,
                pressure: 1013,
                sea_level: 1013,
                grnd_level: 952,
                humidity: 86,
                temp_kf: 0,
                time: '21:00'
            }
        },
        segments: [
            {
                temperatureInCelsius: 14,
                temperatureInFahrenheit: 57.2,
                time: '00:00'
            },
            {
                temperatureInCelsius: 13,
                temperatureInFahrenheit: 55.4,
                time: '03:00'
            },
            {
                temperatureInCelsius: 17,
                temperatureInFahrenheit: 62.6,
                time: '06:00'
            },
            {
                temperatureInCelsius: 22,
                temperatureInFahrenheit: 71.6,
                time: '09:00'
            },
            {
                temperatureInCelsius: 25,
                temperatureInFahrenheit: 77,
                time: '12:00'
            },
            {
                temperatureInCelsius: 25,
                temperatureInFahrenheit: 77,
                time: '15:00'
            },
            {
                temperatureInCelsius: 21,
                temperatureInFahrenheit: 69.8,
                time: '18:00'
            },
            {
                temperatureInCelsius: 17,
                temperatureInFahrenheit: 62.6,
                time: '21:00'
            }
        ],
        temperature: 2342.84,
        minTemperature: 2342.84,
        maxTemperature: 2342.84,
        count: 8,
        date: '23-07-2021',
        dateString: '23 Jul 21'
    }
};