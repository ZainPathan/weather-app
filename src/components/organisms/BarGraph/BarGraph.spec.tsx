import React from 'react';
import renderer from 'react-test-renderer';
import BarGraph from './BarGraph';
import { mockWeatherData } from '../../../core/mock/mockWeatherData';

describe('BarGraph component tests', () => {
    const mock = {
        currentCardDate: '19-07-2021',
        temperatureUnit: 'fahrenheit',
        weatherData: mockWeatherData
    }

    it('renders properly', () => {
        const tree = renderer
            .create(
                <BarGraph
                    weatherData={mock.weatherData}
                    currentCardDate={mock.currentCardDate}
                    temperatureUnit={mock.temperatureUnit}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})
