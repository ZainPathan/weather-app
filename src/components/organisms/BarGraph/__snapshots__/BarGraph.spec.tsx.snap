// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarGraph component tests renders properly 1`] = `
<div
  className="recharts-responsive-container"
  style={
    Object {
      "height": "100%",
      "maxHeight": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "width": "100%",
    }
  }
/>
`;
